### DISCUSSION TITLE
BitOps Meeting (04 Jan 2023): BitOps Actions, Community Metrics, CLI Ops Repo Generator, v2.4.0

### DISCUSSION THREAD

**Discussion by arm4b** [bitovi/bitops#384] [Announcements]
We invite everyone (developers, adopters, or just curious people) to join the bi-weekly BitOps Community `1 hour` Meeting on Wednesday, `04 Jan 2023, 12:00 PM US East` to discuss the ongoing project developments, updates, and plans, share use cases and stories or ask/answer questions about BitOps tool.

See https://github.com/bitovi/bitops/discussions/288 to find the meeting link and schedule.

### Follow-ups from the previous #370 Meeting
- BitOps landing page (@mickmcgrath13 @armab)
- BitOps 1-3min intro Video (@PhillypHenning)
- BitOps Twitch update (@PhillypHenning @LeoDiazL)

### BitOps GH Actions: StackStorm
Current state and progress of using BitOps in GH Actions (StackStorm) and what's next.

### BitOps Community Metrics
@armab to share the current community metrics dashboard draft.

### Planning: BitOps OpsRepo Generator CLI in Golang
Discuss the repository, some first tasks, feature set, CI, tests, releases, artifacts building, publishing and distribution, how do we do the work, share the Golang knowledge, etc.

### v2.4.0 Release Planning
See https://github.com/bitovi/bitops/milestone/10 with the (5) issues that were transferred from the previous `v2.3.0`
- DeCLI? https://github.com/bitovi/bitops/issues/230
- New Cloud Provider(s)?

**Comment by mickmcgrath13** [comment]
# 2023-01-04 - Meeting notes

## Follow Ups
- Landing page - higher quality, polish, and content
- docs landing page, too
- 1-3 minute video
  - foresight video

## GitHub Actions
- stackstorm
  - promotion (bitovi first, st2 later).  
  - Call it "beta"
  - should have TF plan in PRs
  - general marketing - present the github action, not bitops
- docker to ec2

## Community Metrics
- @armab showed kibana dashboard from chaoss metrics
- is chaoss just a backend?
  - elasticsearch, mariadb, hatstall (kibana project), mordred (data source pulling)
- can we create and publish pre-defined dashboards?
- used an existing dashboard? - not there quite yet. on the list
- how do you configure a new "data source"?
  - added manually, could export into JSON
- how do you create a new "data source"?
  - there are docs
- how's this currently configured?
  - ec2 instance (ssh'd) with docker compose
- What are next steps?
  - change default passwords
  - read only dashboards to the public
  - save data
  - create new data source
  - share created dashboards for others to use on their projects

**Comment by arm4b** [comment]
Documented a [new feature to integrate with CDEvents #385](https://github.com/bitovi/bitops/issues/385) in the future.
Something we unintentionally covered during the call.

CDEvents in the new specification created by the CD Foundation that allows exposing useful data about the CD related to builds, time, deployments, failures, time to recovery, and so on.

Eventually, that transforms into a [DORA metrics](https://cloud.google.com/blog/products/devops-sre/using-the-four-keys-to-measure-your-devops-performance) as it helps answer how fast-moving and "Elite" is the DevOps team with the real metrics and measures (Deployment Frequency, Lead Time for Changes, Change Failure Rate, How many Deployments per week, etc).

This is good for businesses, enterprises, and any serious development teams.

We may want to integrate with that in the future as the specification matures (currently `v0.1.0`) with more adoption.
